scripts/install-snowflake-driver.sh

```diff
@@ -1,42 +1,47 @@
 #!/usr/bin/env bash
 
+# Install the Snowflake ODBC driver on a Debian-based system.
+# https://developers.snowflake.com/odbc/
+
 set -xeuo pipefail
 
-## Specify your organization - account name as the account identifier
-SFACCOUNT=${SFACCOUNT:-myorganization-myaccount}
-VERSION="3.3.2"
-FILE="snowflake-odbc-${VERSION}.x86_64.deb"
-URL="https://sfc-repo.snowflakecomputing.com/odbc/linux/${VERSION}/${FILE}"
-SHA256="fdcf83aadaf92ec135bed0699936fa4ef2cf2d88aef5a4657a96877ae2ba232d"
+if [ ! -f /etc/debian_version ]; then
+  echo "This script is only intended for Debian-based systems"
+  exit 1
+fi
 
-if [[ -f "${FILE}" && $(sha256sum "${FILE}" | cut -f1 -d' ') == "${SHA256}" ]]; then
-  echo "snowflake package already downloaded"
+VERSION="3.4.1"
+BASE_URL="https://sfc-repo.snowflakecomputing.com/odbc"
+ARCH="$(uname -m)"
+FILE="snowflake-odbc-${VERSION}.${ARCH}.deb"
+if [ "${ARCH}" == x86_64 ]; then
+  URL="${BASE_URL}/linux/${VERSION}/${FILE}"
+  SHA256="a96fcc89a3d3f2d16fc492976a01fefae57029bcd2e238ff4eff5a6693fa4f74"
+elif [ "${ARCH}" == aarch64 ]; then
+  URL="${BASE_URL}/linux${ARCH}/${VERSION}/${FILE}"
+  SHA256="c9f5c60ace416640683693037d5949aefea9555b9aa62501b6b6c692d140989c"
 else
-  echo "downloading snowflake package"
-  wget -nc "$URL"
+  echo "Unsupported architecture: ${ARCH}"
+  exit 1
 fi
 
-function configure() {
-  ODBC_INST_LIB=/usr/lib/x86_64-linux-gnu/libodbcinst.so
+apt-get -qq update && apt-get install -yqq curl
+curl -fsSL -O --no-clobber --retry 5 "$URL"
+echo "$SHA256 *$FILE" | sha256sum -c
+apt-get -qq update
+apt-get install -yqq unixodbc-dev unixodbc odbcinst curl libreadline8
+apt-get install -yqq "./${FILE}"
 
-  sed -i -e "s#^ODBCInstLib=.*#ODBCInstLib=$ODBC_INST_LIB#" /usr/lib/snowflake/odbc/lib/simba.snowflake.ini
+# Remove references to SF_ACCOUNT from the odbc.ini file.
+# It will be configured dynamically.
+sed -i '/^SERVER/d' /etc/odbc.ini
+sed -i '/^ACCOUNT/d' /etc/odbc.ini
 
-  sed -i -e "s#SF_ACCOUNT#${SFACCOUNT}#" /etc/odbc.ini
+ODBC_INI_CONTENT="[ODBC Data Sources]
+snowflake = SnowflakeDSIIDriver"
 
-  cat >>/etc/odbc.ini  <<EOF
-[ODBC Data Sources]
-snowflake = SnowflakeDSIIDriver
+if ! grep -Fxq "$ODBC_INI_CONTENT" /etc/odbc.ini; then
+    cat >>/etc/odbc.ini <<EOF
+$ODBC_INI_CONTENT
 EOF
-}
-
-if ! dpkg -l snowflake-odbc 1>/dev/null 2>/dev/null ; then
-  apt update && apt install -yyq unixodbc-dev odbcinst
-  dpkg -i "${FILE}"
-  apt install -f
-
-  configure
-
-  echo "installed and configured snowflake"
-else
-  echo "snowflake odbc already installed; not attempting to configure it"
```

Comment:
I think that removing this will make the CI job for the SF bridge a bit slower, because it'll start to install this without really using it.

Reply:
D'oh, ignore me, I was thinking of a check that lives in `run.sh`. :speak_no_evil: :see_no_evil:

Reply:
Too late! Added early exit from this script if the package is already installed.